### DISCUSSION TITLE
Outcome then Exposure; reverse engineer the system

### DISCUSSION THREAD

**Discussion by ghost** [MRCIEU/TwoSampleMR#366] [Q&A] [answered]
Below I have shown a standard script I've been using for performing MR.
I was wondering if there was a way to "reverse engineer" the system, looking to see if there is any way to see how the outcome may affect the exposure?

Many thanks for your help in advance!

**Standard script**
exposure_dat <- format_gtex_eqtl(gtex_eqtl[c(192866	), ])
exposure_dat <- clump_data(exposure_dat)
outcome_dat <- extract_outcome_data(exposure_dat$SNP, _c('_example_'),_ proxies = 1, rsq = 0.8, align_alleles = 1, palindromes = 1, maf_threshold = 0.3)
dat <- harmonise_data(exposure_dat, outcome_dat, action = 2)
mr_results <- mr(dat)
mr_results

**Comment by remlapmot** [comment, marked as answer]
If you want to swap the outcome and exposure then that is known as a bidirectional/reciprocal/reverse direction Mendelian randomization analysis. For that you need to change your set of genotypes to be those known to be associated with your current outcome (which has become your exposure).

The original bidirectional MR paper was by Timpson et al. [here](https://doi.org/10.1038/ijo.2010.137). But there are now many other papers using the approach.

**Reply by ghost** [reply]
Thanks for the response, really helpful!

I had come up with a method, but unsure if it actually stands, what are your thoughts?

**The Method**
I'm using MR analysis to look at the relationship of SNP -> A -> B, now I want to see if SNP -> B -> A is also true. 
I used the same code as SNP -> A -> B, but stopped after data harmonisation. 

I then changed the column names of exposure to outcome and outcome to exposure and ran MR using the data I renamed. 
The package I'm using is TwoSampleMR, I'm wondering if this method is acceptable/ will be giving me the right results?
